controllers/validator_test.go

```diff
@@ -806,6 +850,62 @@ func TestDPAReconciler_ValidateDataProtectionCR(t *testing.T) {
 			wantErr: true,
 			want:    false,
 		},
+		{
+			name: "given valid DPA CR AWS with VSL credentials referencing a non-existent secret",
+			dpa: &oadpv1alpha1.DataProtectionApplication{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-DPA-CR",
+					Namespace: "test-ns",
+				},
+				Spec: oadpv1alpha1.DataProtectionApplicationSpec{
+					BackupLocations: []oadpv1alpha1.BackupLocation{
+						{
+							Velero: &v1.BackupStorageLocationSpec{
+								Provider: "velero.io/aws",
+								Credential: &corev1.SecretKeySelector{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "cloud-credentials",
+									},
+									Key:      "cloud",
+									Optional: new(bool),
+								},
+							},
+						},
+					},
+					SnapshotLocations: []oadpv1alpha1.SnapshotLocation{
+						{
+							Velero: &v1.VolumeSnapshotLocationSpec{
+								Provider: "velero.io/aws",
+								Credential: &corev1.SecretKeySelector{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "bad-credentials",
+									},
+									Key:      "bad-key",
+									Optional: new(bool),
```

Comment:
this is new to me.... wonder what this do.

Reply:
It's just a reference to a secret that doesn't exist, which should cause a validation error. That is the most direct test for the request in OADP-1526.

Reply:
I mean the `optional: new (bool)`

Reply:
Oh, okay. I'm not too sure either, I copied it from another test.